cmd/clusterctl/client/cluster/template.go

```diff
@@ -51,6 +52,7 @@ type templateClient struct {
 	configClient        config.Client
 	gitHubClientFactory func(configVariablesClient config.VariablesClient) (*github.Client, error)
 	processor           yaml.Processor
+	httpClient          http.Client
```

Comment:
```suggestion
	httpClient          *http.Client
```

Reply:
Also, in the `newTemplateClient` function you will need to set the httpClient to `http.DefaultClient`.

(cannot mark the change here in this PR)

Reply:
Why do I have to add an `http.DefaultClient`? I am not using this function in the `Test_templateClient_getRawUrlFileContent` right?

Reply:
If you do not `t.httpClient` is never set and in l.219 will throw a nil pointer exception. 

>Why do I have to add an http.DefaultClient? I am not using this function in the Test_templateClient_getRawUrlFileContent right?

You are setting the httpClient value in l.259 in the test file. You also need to set the value in `newTemplateClient` so that the command actually runs outside of the unit tests.

Reply:
Yep make sense!